src/TextInputWithIcon/TextInputWithIcon.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import { IconType } from 'react-icons/src'
+
+import { TextInput, TextInputProps, OmitOnChangeHTMLInputAttributes } from '../TextInput/TextInput'
+import { matchThemeProp, InputProps, defaultStyledProps } from '../utils/styled'
+import { Icon } from '../Icon'
+
+export interface TextInputWithIconProps extends TextInputProps {
+  icon: React.ReactElement<null, IconType>
+  iconLeft?: boolean
+}
+
+interface InputIconProps extends InputProps {
+  iconLeft?: boolean
+  disabled?: boolean
+}
+
+const StyledContainer = styled.div `
+  position: relative;
+  display: flex;
+  justify-content: column;
+  align-items: center;
```

Comment:
if you add the following here, the icon will not stop interactions with the input below (i.e. in SelectInput now you can't click on the icon to open the dropdown):

```
${Icon} {
  pointer-events: none;
}

Reply:
Good one!